### PR TITLE
Sort actors by number of movies

### PR DESCRIPTION
**Pull request by Pedrock**
Option to sort actors by the number of movies in the library.
Feature Request: http://forum.kodi.tv/showthread.php?tid=238563

### CODE REVIEW

File: xbmc/video/VideoDatabase.cpp [hunk 1]
```diff
@@ -5572,6 +5575,10 @@ bool CVideoDatabase::GetPeopleNav(const std::string& strBaseDir, CFileItemList&
           if (g_passwordManager.IsDatabasePathUnlocked(std::string(m_pDS->fv("path.strPath").get_asString()),*CMediaSourceSettings::GetInstance().GetSources("video")))
             mapActors.insert(std::pair<int, CActor>(idActor, actor));
         }
+        else
+        {
```
Comment (Montellese) [inline xbmc/video/VideoDatabase.cpp]: No curly braces on single line blocks.

**Comment by Montellese** [comment]
This is something I had on my TODO list as well but I wanted to do it in a more generic way and call it "Sort by relevance". That would allow us to reuse the same field in `CVideoInfoTag` for other things as well (e.g. sort genres by number of movies etc).

File: xbmc/utils/DatabaseUtils.h [hunk 2]
```diff
@@ -129,7 +129,8 @@ typedef enum {
   FieldDied,
   FieldStereoMode,
   FieldUserRating,
-  FieldMax
+  FieldMax,
+  FieldMovieCount
```
Comment (Razzeee) [inline xbmc/utils/DatabaseUtils.h]: FieldMax should always be the last element.

**Comment by Pedrock** [comment]
@Montellese @Razzeee I have fixed what you mentioned.
I also added a new language string.

File: xbmc/utils/LabelFormatter.cpp [hunk 3]
```diff
@@ -258,6 +258,9 @@ std::string CLabelFormatter::GetMaskContent(const CMaskString &mask, const CFile
   case 'C': // programs count
     value = StringUtils::Format("%i", item->m_iprogramCount);
     break;
+  case 'c': // relevance
```
Comment (Razzeee) [inline xbmc/utils/LabelFormatter.cpp]: Please also add this to the comment above

Comment (Pedrock) [inline xbmc/utils/LabelFormatter.cpp]: @Razzeee What do you mean?

Comment (Razzeee) [inline xbmc/utils/LabelFormatter.cpp]: The useage of 'c' should be mentioned in the methods comments:
https://github.com/Pedrock/xbmc/blob/ActorSorting/xbmc/utils/LabelFormatter.cpp#L103

Comment (Pedrock) [inline xbmc/utils/LabelFormatter.cpp]: @Razzeee Done!

**Comment by Razzeee** [comment]
I get the feeling, that that concept of a very loose type of sorting could come back and bite us as it's hard to follow, if you don't know what's happening.
We need at least one place to document it in the code, as there is a place where it "breaks" from Relevance to "Number of Movies".

Also is this strictly movies? Does this work for tvshows?

**Comment by Pedrock** [comment]
@Razzeee I hadn't tested it for tvshows because I hadn't any but it seems to be broken. I was expecting it to not show the new ordering but it does because it is the same menu and it shows -1 for all of the actors.
I am working on a fix. It should show the number of tvshows and not the number of episodes, right? I am going to need to add to the sql query.
Do you want me to change it back from relevance to something else? Any suggestions?

**Comment by Razzeee** [comment]
Yes, it should show the number of tv shows the person is in.
Maybe appearances is more generic and at the same time pointing in the right direction?

Let's wait for @Montellese regarding relevance.
But I would say, keep it this way but let's add some more comments, so anyone reading it will know what it's for. (if you don't want to do that, I can add this after this get's merged)

**Comment by Pedrock** [comment]
@Razzeee I will change the string to "Appearances" then.
I will also fix the tvshows and add some more comments.

**Comment by Pedrock** [comment]
@Razzeee It is now working for tv shows too. I faked some tvshow files in order to get some actors, including actors with appearances in multiple shows. I tested everything in both "master locked" and unlocked modes. I also added some comments as you suggested.

**Comment by Razzeee** [comment]
Thanks, let's wait for @Montellese to check and some others to have a look.

**Comment by Montellese** [comment]
`Appearances` sounds better to me than what you had before. Nice work.

**Comment by Pedrock** [comment]
@Montellese Any more feedback? Is this getting merged?

**Comment by Montellese** [comment]
Sorry my "Nice work" was meant as "Should be good to go".

**Comment by Razzeee** [comment]
jenkins build this please

**Comment by Razzeee** [comment]
jenkins build and merge
